packages/shared/src/components/fields/CustomSwitch.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { ReactElement, useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import styles from './CustomSwitch.module.css';
+import classed from '../../lib/classed';
+
+export interface ContentsSwitchProps {
+  className?: string;
+  inputId: string;
+  name: string;
+  checked?: boolean;
+  onToggle?: () => unknown;
+  leftContent: React.FC<{ className?: string }> | string;
+  rightContent: React.FC<{ className?: string }> | string;
```

Comment:
Did we not say to not use FC? I remember some discussion about this a while ago, but happy to leave this for now and discuss in gathering

Reply:
This was just from the original file itself, but yes I think this can be a good discussion in the Gathering.

![image](https://user-images.githubusercontent.com/13744167/194592671-6261f1e6-b0e7-4c93-897f-10c4e9335201.png)